src/plugins/instrument_fetch.js

```diff
@@ -189,9 +191,11 @@ class InstrumentFetch {
             // Add Open-Tracing headers
             const headersCarrier = {};
             tracer.inject(span.context(), opentracing.FORMAT_HTTP_HEADERS, headersCarrier);
-            Object.keys(headersCarrier).forEach((key) => {
-                if (!request.headers.get(key)) request.headers.set(key, headersCarrier[key]);
-            });
+            if (self._shouldAddHeadersToRequest(tracer, request.url)) {
```

Comment:
Is it possible to move this up so that we don't have to create `headersCarrier` or make the call to `tracer.inject`?

Reply:
Makes sense will do.